smart-contracts/scripts/deploy.js

```diff
@@ -0,0 +1,78 @@
+const { constants } = require('hardlydifficult-ethereum-contracts')
+const { ethers, upgrades } = require('hardhat')
+
+const { getNetworkName } = require('../helpers/network')
+const { addDeployment } = require('../helpers/deployments')
+
+async function main() {
```

Comment:
Can you confirm this only deploys the _latest_ versions?

Reply:
yes, currently no upgrades. It will deploy only the latest

Reply:
the `addDeployment` helpers are to keep track of what has been deployed (for the tests mostly)

Reply:
Perfect! Makese sense